lib/screens/home/home.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:flutter/material.dart';
+import 'package:innopolis_feedback/services/auth.dart';
+import 'package:innopolis_feedback/shared/loading.dart';
+import 'package:innopolis_feedback/ta_course_page.dart';
+import 'package:innopolis_feedback/data.dart';
+import 'package:innopolis_feedback/ui/app_bar.dart';
+import 'package:innopolis_feedback/ui/list_item.dart';
+
+class MyHomePage extends StatefulWidget {
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  List<Year> years;
+  Year selectedYear;
+  Course selectedCourse;
+  TA selectedTA;
+
+  List currentList;
+  Function currentBuilder;
+
+  goBack() {
+    if (selectedCourse != null)
+      return setState(() {
+        selectedCourse = null;
+        selectYear(selectedYear);
+      });
+    if (selectedYear != null)
+      return setState(() {
+        selectedYear = null;
+      });
+  }
+
+  selectYear(Year year) async {
+    print('Selected year: ' + year.name);
+    currentList = await getCoursesByYear(year);
+    setState(() {
+      selectedYear = year;
+      currentBuilder = courseItemBuilder;
+    });
+  }
+
+  selectCourse(Course course) async {
+    print('Selected course: ' + course.name);
+
+    currentList = await getTAs(course);
+    setState(() {
+      selectedCourse = course;
+      currentBuilder = taItemBuilder;
+    });
+  }
+
+  selectTA(TA ta) {
+    print('Selected TA: ' + ta.name);
+    String title = ta.name + ' - ' + selectedCourse.name;
+    Navigator.push(
+        context,
+        MaterialPageRoute(
+            builder: (context) =>
+                TaCoursePage(title, ta.id, selectedCourse.id)));
+  }
+
+  Widget yearItemBuilder(Year year) {
+    return ListItem(
+      title: year.name,
+      onTap: () => selectYear(year),
+    );
+  }
+
+  Widget courseItemBuilder(Course course) {
+    return ListItem(
+      title: course.name,
+      onTap: () => selectCourse(course),
+    );
+  }
+
+  Widget taItemBuilder(TA ta) {
+    return ListItem(
+      title: ta.name,
+      onTap: () => selectTA(ta),
+      leading: CircleAvatar(
+        backgroundColor: Colors.deepPurple[100],
+        radius: 18,
+        child: Opacity(
+          opacity: 0.9,
+          child: Icon(
+            Icons.person,
+            color: Colors.white,
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: CustomAppBar(
+        title: 'Innopolis Feedback',
+        actions: <Widget>[
+          FlatButton.icon(
+            icon: Icon(Icons.person),
+            label: Text('Sign out?'),
+            onPressed: () => AuthService().signOut(),
+          ),
+        ],
+        goBackIconVisible: selectedYear != null || selectedCourse != null,
```

Comment:
Nope. Not good.

![image](https://user-images.githubusercontent.com/8573325/102121838-a9645800-3e55-11eb-9eee-5ad97a0ed557.png)

Reply:
Suggestion 1: Add the back button.
Suggestion 2: Remove the sign out button.

Reply:
Sign out button is remove at upd-design-2